llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -15877,6 +15877,12 @@ SITargetLowering::getRegForInlineAsmConstraint(const TargetRegisterInfo *TRI_,
             RC = TRI->getAGPRClassForBitWidth(Width);
           if (RC) {
             Reg = TRI->getMatchingSuperReg(Reg, AMDGPU::sub0, RC);
+            if (Reg == 0U) {
```

Comment:
```suggestion
            if (!Reg ) {
```

Eventually this should stop returning an unsigned